webapp/src/ts/modals/edit-message-group/edit-message-group.component.ts

```diff
@@ -0,0 +1,154 @@
+import { AfterViewChecked, AfterViewInit, Component, OnDestroy, OnInit } from '@angular/core';
+import * as moment from 'moment';
+import { BsModalRef } from 'ngx-bootstrap/modal';
+import { v4 as uuid } from 'uuid';
+
+import { EditGroupService } from '@mm-services/edit-group.service';
+import { SettingsService } from '@mm-services/settings.service';
+import { MmModalAbstract } from '@mm-modals/mm-modal/mm-modal';
+
+@Component({
+  selector: 'edit-message-group',
+  templateUrl: './edit-message-group.component.html',
+})
+export class EditMessageGroupComponent extends MmModalAbstract
+  implements AfterViewInit, OnInit, AfterViewChecked, OnDestroy {
+
+  constructor(
+    bsModalRef:BsModalRef,
+    private editGroupService:EditGroupService,
+    private settingsService:SettingsService,
+  ) {
+    super(bsModalRef);
+  }
+
+  model:any = { group: { rows: [] } };
+
+  private shouldInitDatePickers = false;
+  private settingsPromise;
+  private datePickers = {};
+
+  ngOnInit() {
+    this.getSettings();
+    // task_id variable is only used within this component, to uniquely identify each task in the group
+    this.model?.group?.rows?.forEach(row => row.task_id = uuid());
+  }
+
+  private getSettings() {
+    this.settingsPromise = this.settingsService.get();
+  }
+
+  private getNextHalfHour() {
+    const time = moment()
+      .second(0)
+      .millisecond(0);
+    if (time.minute() < 30) {
+      time.minute(30);
+    } else {
+      time.minute(0);
+      time.add(1, 'hours');
+    }
+    return time;
+  }
+
+  trackBy(index, task) {
+    return task.task_id;
+  }
+
+  private initDatePickers() {
+    this.settingsPromise.then((settings:any) => {
+      $('#edit-message-group input.datepicker').each((index, element) => {
+        const taskId = $(element).data('task-id');
```

Comment:
In the past when having more than one call to `$(element)` in a block, I used to create a local variable to assign the value:

```js
let $element = $(element);
```

And then used the variable `$element` instead of `$(element)`, but it has more sense when `element` is a string with a CSS expression than when is a HTML element, though not sure how much objects and time jQuery spend each time we do again `$(element)`.

Reply:
Changed, thanks!